src/Microsoft.AspNetCore.Blazor.Server/AutoRebuild/AutoRebuildExtensions.cs

```diff
@@ -19,7 +21,37 @@ internal static class AutoRebuildExtensions
         private static string[] _includedSuffixes = new[] { ".cs", ".cshtml", "index.html" };
         private static string[] _excludedDirectories = new[] { "obj", "bin" };
 
-        public static void UseAutoRebuild(this IApplicationBuilder appBuilder, BlazorConfig config)
+        public static void UseHostedAutoRebuild(this IApplicationBuilder appBuilder, BlazorConfig config, string hostAppContentRootPath)
+        {
+            var isFirstFileWrite = true;
+            WatchFileSystem(config, () =>
```

Comment:
You'll have to excuse my ignorance, but won't `FileSystemWatcher` be GCed?

Reply:
Good point! Now accounted for.